src/mesh/distributed_mesh.C

```diff
@@ -77,41 +85,22 @@ DistributedMesh & DistributedMesh::operator= (DistributedMesh && other_mesh)
   this->UnstructuredMesh::operator=(std::move(other_mesh));
 
   // The remaining work to move a DistributedMesh
-  _is_serial = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).is_serial());
-  _is_serial_on_proc_0 = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).is_serial_on_zero());
+  _is_serial = reinterpret_cast<DistributedMesh &>(other_mesh).is_serial();
+  _is_serial_on_proc_0 = reinterpret_cast<DistributedMesh &>(other_mesh).is_serial_on_zero();
 
-  // These are now obtained in DistributedMesh::move_nodes_and_elements
-  //_n_nodes = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).n_nodes());
-  //_n_elem = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).n_elem());
+  _max_node_id = reinterpret_cast<DistributedMesh &>(other_mesh).max_node_id();
+  _max_elem_id = reinterpret_cast<DistributedMesh &>(other_mesh).max_elem_id();
 
-  _max_node_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).max_node_id());
-  _max_elem_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh).max_elem_id());
-
-  _next_free_local_node_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh)._next_free_local_node_id);
-  _next_free_local_elem_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh)._next_free_local_elem_id);
-  _next_free_unpartitioned_node_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh)._next_free_unpartitioned_node_id);
-  _next_free_unpartitioned_elem_id = std::move(reinterpret_cast<DistributedMesh &>(other_mesh)._next_free_unpartitioned_elem_id);
+  _next_free_local_node_id = reinterpret_cast<DistributedMesh &>(other_mesh)._next_free_local_node_id;
```

Comment:
Actually, I think there's no need for any casts here, even on the non-MeshBase data.  other_mesh is already an lvalue (despite being passed in as an rvalue; basically it didn't "have a name" when it got passed in but we "gave it a name" when we made it a parameter argument).